src/scripts/main.js

```diff
@@ -1,3 +1,54 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+function waitForClick(element) {
+  return new Promise((resolve) => {
+    element.addEventListener('click', () => resolve('Element was clicked'), {
+      once: true,
+    });
+  });
+}
+
+function successMessage() {
+  const div = document.createElement('div');
+
+  div.innerText = 'Promise was resolved!';
+  div.className = 'message';
+
+  return div;
+}
+
+function errorMesssage(err) {
```

Comment:
There is a typo in the function name `errorMesssage`. It should be `errorMessage` to maintain consistency and readability.